### PR TITLE
Fixed an issue where formally finite values were approximated with nu…

### LINKED ISSUE #12833: kuiper_two returns nan for fpp value
### Description

astropy.stats.kuiper_two may return np.nan value for fpp in some cases.
### Expected behavior

false positive probability should not be of type NaN
### Actual behavior

Function return NaN type for false positive probability in some cases.

### Steps to Reproduce

See the two arrays that produce the issue here https://github.com/vyktran/kuiper_two

```python
# Put your Python code snippet here.
import numpy as np
from astropy.stats import kuiper_two

arr0 = np.load('arr0.npy')
arr1 = np.load('arr1.npy')
kuiper_two(arr0,arr1)
```
image

### System Details

macOS-12.1-x86_64-i386-64bit
Python 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 20:33:18) 
[Clang 11.1.0 ]
Numpy 1.21.4
pyerfa 2.0.0.1
astropy 5.0.1
Scipy 1.7.3
Matplotlib 3.5.0

### PR DESCRIPTION
**Pull request by adam-broussard**
…mpy.inf, producing NaN probabilities

### Description

This pull request fixes the issue where returned probabilities are NaNs.  The [combination calculation](https://github.com/astropy/astropy/blob/0867721075391c359f7c5e7ab6ac7d92b037b64e/astropy/stats/funcs.py#L1364) can return large values for distributions with many points, which are approximated as numpy.inf.  When these are multiplied by zero, NaNs are returned.  I have implemented a check to modify array values where numpy.inf is multiplied by zero to give zero instead because all values are formally finite.

Fixes #12833 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Hello and thanks! Would need a change log and test as well.

**Comment by adam-broussard** [comment]
@pllim I think this is now ready to go!  Thanks for helping me through my first PR with Astropy :)

**Comment by pllim** [comment]
Thank YOU, @adam-broussard ! Let's wait for a review from subpackage maintainer. Once that is complete, this would still need a squash but not yet.

**Review by larrybradley — APPROVED** [review]
LGTM, thanks.

**Comment by pllim** [comment]
@adam-broussard , can you please squash all your commits into one commit? Thanks!

Thanks for the review, @larrybradley !

**Comment by adam-broussard** [comment]
@pllim done!

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**
